tests/integration/pilot/common/routing.go

```diff
@@ -1955,3 +1955,38 @@ func serverFirstTestCases(apps *EchoDeployments) []TrafficTestCase {
 
 	return cases
 }
+
+func websocketTestCases(apps *EchoDeployments) []TrafficTestCase {
+	cases := make([]TrafficTestCase, 0)
+	clients := apps.PodA
+	destination := apps.PodC[0]
+	h2UpgradeDR := func(dst string) string {
+		return fmt.Sprintf(`apiVersion: networking.istio.io/v1beta1
+kind: DestinationRule
+metadata:
+  name: %s
+spec:
+  host: %s
+  trafficPolicy:
+    connectionPool:
+      http:
+        h2UpgradePolicy: UPGRADE
```

Comment:
I think we want to test more than just upgrade, also client sending WS over h2 directly?

Reply:
I don't think `gorilla/websockets` (what echo client uses) supports websockets over H2? [https://github.com/gorilla/websocket/issues/417](https://github.com/gorilla/websocket/issues/417)

Reply:
For context this was more to directly test the broken case in [https://github.com/istio/istio/issues/33534](https://github.com/istio/istio/issues/33534)